cumulusci/core/config/BaseProjectConfig.py

```diff
@@ -3,32 +3,106 @@
 
 import cumulusci
-from cumulusci.core.config import BaseTaskFlowConfig
-from cumulusci.core.exceptions import ConfigError
-from cumulusci.core.exceptions import DependencyResolutionError
-from cumulusci.core.exceptions import KeychainNotFound
-from cumulusci.core.exceptions import ServiceNotConfigured
-from cumulusci.core.exceptions import ServiceNotValid
+from cumulusci.core.config import BaseTaskFlowConfig, MergedConfig
+from cumulusci.core.exceptions import (
+    ConfigError,
+    DependencyResolutionError,
+    KeychainNotFound,
+    ServiceNotConfigured,
+    ServiceNotValid,
+    NotInProject,
+    ProjectConfigNotFound,
+)
 from cumulusci.core.github import get_github_api
 
 
 class BaseProjectConfig(BaseTaskFlowConfig):
     """ Base class for a project's configuration which extends the global config """
 
-    search_path = ["config"]
+    config_filename = "cumulusci.yml"
 
-    def __init__(self, global_config_obj, config=None):
+    def __init__(self, global_config_obj, config=None, *args, **kwargs):
         self.global_config_obj = global_config_obj
         self.keychain = None
+        self._repo_info = None
         if not config:
             config = {}
+
+        # Initialize the dictionaries for the individual configs
+        self.config_project = {}
+        self.config_project_local = {}
+        self.config_additional_yaml = {}
+
+        # optionally pass in a kwarg named 'additional_yaml' that will
+        # be added to the YAML merge stack.
+        self.additional_yaml = None
+        if "additional_yaml" in kwargs:
+            self.additional_yaml = kwargs.pop("additional_yaml")
+
         super(BaseProjectConfig, self).__init__(config=config)
 
+    @property
+    def config_project_local_path(self):
+        path = os.path.join(self.project_local_dir, self.config_filename)
+        if os.path.isfile(path):
+            return path
+
+    def _load_config(self):
+        """ Loads the configuration from YAML, if no override config was passed in initially. """
+
+        if (
+            self.config
+        ):  # any config being pre-set at init will short circuit out, but not a plain {}
+            return
+
+        # Verify that we're in a project
+        repo_root = self.repo_root
+        if not repo_root:
+            raise NotInProject(
+                "No git repository was found in the current path. You must be in a git repository to set up and use CCI for a project."
+            )
+
+        # Verify that the project's root has a config file
+        if not self.config_project_path:
+            raise ProjectConfigNotFound(
+                "The file {} was not found in the repo root: {}. Are you in a CumulusCI Project directory?".format(
+                    self.config_filename, repo_root
+                )
+            )
+
+        # Load the project's yaml config file
+        with open(self.config_project_path, "r") as f_config:
+            project_config = yaml.safe_load(f_config)
```

Comment:
I wonder if we should use the ordered load in here too, to be safely backwards compatible in Python 2. It's conceivable that someone could be using a yaml dict option for a task and relying on its order being preserved.